src/components/NewMovie/NewMovie.js

```diff
@@ -1,13 +1,127 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 
 export class NewMovie extends Component {
-  state = {};
+  state = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+    submitDisabled: true,
+  };
+
+  onInputChange = (value, id) => {
+    this.setState(state => ({
+      [id]: value,
+      submitDisabled: !(
+        state.title !== ''
+          && state.imgUrl !== ''
+          && state.imdbUrl !== ''
+          && state.imdbId !== ''
+      ),
+    }));
+  }
+
+  clearForm() {
+    this.setState({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+      submitDisabled: true,
+    });
+  }
 
   render() {
+    const { onAdd } = this.props;
+    const {
+      title, description, imgUrl, imdbUrl, imdbId, submitDisabled,
+    } = this.state;
+
     return (
-      <form>
-        Put the form here
+      <form
+        name="newMovie"
+        className="field"
+        onSubmit={(event) => {
+          event.preventDefault();
+          onAdd(title, description, imgUrl, imdbUrl, imdbId);
+          this.clearForm();
+        }}
+      >
+        <label htmlFor="title">Movie title </label>
+        <div className="control">
+          <input
+            className="input"
+            type="text"
+            id="title"
+            value={title}
+            onChange={
+              event => this.onInputChange(event.target.value, event.target.id)
+            }
```

Comment:
```suggestion
            onChange={this.onInputChange}
```

it's better to pass func like this and then destructure all the needed props from event.target, cause now you create 5 new functions on each rerender